cleanlab/classification.py

```diff
@@ -512,31 +512,29 @@ class 0, 1, ..., K-1. `pred_probs` should have been computed using 3 (or
                 if self.verbose:
                     print("Fitting final model on the clean data ...")
             else:
-                if (
-                    "sample_weight" in inspect.getfullargspec(self.clf.fit).args
-                    and "sample_weight" not in self.clf_final_kwargs
-                    and self.noise_matrix is None
-                ):
-                    print(
-                        "Cannot utilize sample weights for final training. To utilize must either specify noise_matrix "
-                        "or have previously called self.find_label_issues() instead of filter.find_label_issues()"
-                    )
-
-                if self.verbose:  # pragma: no cover
+                if self.verbose:
                     if "sample_weight" in self.clf_final_kwargs:
-                        print("Fitting final model on the clean data with custom sample_weight...")
+                        print("Fitting final model on the clean data with custom sample_weight ...")
                     else:
+                        if (
+                            "sample_weight" in inspect.getfullargspec(self.clf.fit).args
+                            and self.noise_matrix is None
+                        ):
+                            print(
+                                "Cannot utilize sample weights for final training. To utilize must either specify noise_matrix "
```

Comment:
nit but would help the user understand what's going on here with the following comment instead:

"Cannot use sample weights for final training!
Details about why this matters: sample weights can be helpful during final training to take into account for the amount of removed data in each class. This helps the learned model to have the correct prior. To use sample weights, you need to either provide the noise_matrix or have previously called self.find_label_issues() instead of filter.find_label_issues() which computes them for you."

Reply:
Note this could be a helpful as warning instead of print if we find out that it actually degrades performance in most real-world use cases. Seems fine as print statement for now.

Reply:
added shortened version of this